core-bundle/src/EventListener/FilterPageTypeListener.php

```diff
@@ -27,11 +39,33 @@ public function __invoke(FilterPageTypeEvent $event): void
             return;
         }
 
-        // Root pages are allowed on the first level only (see #6360)
-        if ($dc->activeRecord->pid > 0) {
-            $event->removeOption('root');
-        } else {
+        // The first level can only have root pages (see #6360)
+        if (!$dc->activeRecord->pid) {
             $event->setOptions(['root']);
+
+            return;
+        }
+
+        $event->removeOption('root');
+
+        $parentType = $this->connection->fetchColumn('SELECT type FROM tl_page WHERE id=?', [$dc->activeRecord->pid]);
```

Comment:
The method is deprecated in favor of the `fetchOne()` method. Did you check if we can use the new method?